packages/core/src/utils/transformers/media/transformMedia.ts

```diff
@@ -20,6 +21,13 @@ export const transformMedia = (
 ) => {
   let value: BoundComponentPropertyTypes;
 
+  // If it is not a deep path and not pointing to the file of the asset,
+  // it is just pointing to a normal field and therefore we just resolve the value as normal field
+  const splitPath = path.split('/');
+  if (!isDeepPath(path) && splitPath[splitPath.length - 2] !== 'file') {
```

Comment:
I read a bit more code and it turns out that `transformMedia()` is only called from one place in the code, and it only accepts an asset as it's argument.  And I was assuming that assets always have `.file` field...  

callsite (note that if-statement which checks guards against `Entry` ) 

![image](https://github.com/contentful/experience-builder/assets/113918657/e8f3bf7a-88dd-4099-ae1a-e5fea81a679f)


and this is grep of usage within entire SDK repo

<img width="1464" alt="image" src="https://github.com/contentful/experience-builder/assets/113918657/a679e4f9-a9c3-447f-b4e8-c7288dd927c6">

Reply:
On my 3rd attempt to read this code, I have inverted some of the if-conditions and correlated with the caller only calling `transformMedia()` on assets and hence came to those conclusions

```js
if (!isDeepPath(path) && splitPath[splitPath.length - 2] !== 'file') {
  // if ( flatPath && notAsset ) {
  // if ( flatPath && isEntry ) {
  // So in theory, if the path was deep, then this would not short-circuit on 'Entry'
  // whilst those conditions are not yet possible, there's no need at all for this more complex logic:
  //  - there's no need to check whether {is entry} or {is asset}, because this function only operates on assets
  //  - there's no need to check whether {is deep} or {is flat}, because what's passed to us is "leaf asset" 
  //    (it's already the actual target asset as per deep path or flat path ) 
  // so this entire if-guard is not needed anymore
    return getBoundValue(asset, path);
  }

```

So it seems we don't need this if-guard at all.